assets/js/src/common/confirm_alert.jsx

```diff
@@ -0,0 +1,60 @@
+import MailPoet from 'mailpoet';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class ConfirmAlert extends React.Component {
+  static propTypes = {
+    title: PropTypes.string,
+    message: PropTypes.string.isRequired,
+    cancelLabel: PropTypes.string,
+    confirmLabel: PropTypes.string,
+    onConfirm: PropTypes.func.isRequired,
+  }
+
+  static defaultProps = {
+    title: MailPoet.I18n.t('confirmTitle'),
+    cancelLabel: MailPoet.I18n.t('cancelLabel'),
+    confirmLabel: MailPoet.I18n.t('confirmLabel'),
+  }
+
+  close = () => {
+    const modalElement = document.getElementById('mailpoet_confirm_alert');
+    ReactDOM.unmountComponentAtNode(modalElement);
+    modalElement.parentNode.removeChild(modalElement);
+  }
+
+  render() {
+    const { title, message, confirmLabel, cancelLabel, onConfirm } = this.props;
+
+    return (
+      <div className="mailpoet_popup" tabIndex="-1">
+        <div className="mailpoet_popup_wrapper">
+          <button className="mailpoet_modal_close" onClick={() => this.close()} />
+          {title &&
+            <div className="mailpoet_popup_title">
+              <h2>{title}</h2>
+            </div>
+          }
+          <div className="mailpoet_popup_body clearfix">
+            <p className="mailpoet_hp_email_label">{message}</p>
+            <button className="button button-secondary" onClick={() => this.close()}>
+              {cancelLabel}
+            </button>
+            <button className="button button-primary" onClick={() => { onConfirm(); this.close(); }}>
+              {confirmLabel}
+            </button>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default function confirmAlert(props) {
+  const modalElement = document.createElement('div');
+  modalElement.className = 'mailpoet_modal_overlay';
+  modalElement.id = 'mailpoet_confirm_alert';
+  document.body.appendChild(modalElement);
+  ReactDOM.render(<ConfirmAlert {...props} />, modalElement);
```

Comment:
We should split this into two different files. One `jsx` file with the component and another with the `ReactDOM.render` stuff. This code ^ limits the usage to only `.js` and we cannot use this component inside another React component.

Reply:
Is there really use case for showing `<ConfirmAlert>` anywhere else then from JS as replacement for `window.confirm`? I understand your reasoning, but it would only create two files, where one is dependant on the other, and the other might not really be used anywhere else. 

It would also either create problem with closing (should it self destroy or let parent know it should close it?) or add complexity where parent must react to any action in `<ConfirmAlert>`.

Reply:
I don't understand why we would use it only in JS files. This could be used anywhere, even in React files. I'd rather have two files because there is no limit of files. 

There can be a function in props that can child call to let the parent know it is closed. So the parent doesn't have to listen to all actions.

Reply:
File split 😉